### DISCUSSION TITLE
How to use the "shift-drag" list feature?

### DISCUSSION THREAD

**Discussion by daspryJ** [piroor/treestyletab#3618] [Q&A] [answered]
Hi, I can't manage to use shift-drag for anything, whatever I drag it into. Does someone have a working exemple of the feature for me to try?
My goal is just to make a tree into a list of URLs. I tried to shift-drag and drop it in gmail web app, online&offline text editing apps, to no avail.
Simple drag and drop does work as intended to detach the tabs from the window, and shift-drag should "create link or bookmark" according to the TST options.

**Comment by piroor** [comment, marked as answer]
Currently TST does not set drag data with `text/plain` data type even if you shift-dragged tabs, this looks to be the reason why nothing is dropped on your case. I've introduced some changes with the commit https://github.com/piroor/treestyletab/commit/758dbdd8d5ab712e600bdf7c67fbd257463a5436 so you'll be able to drop URLs by shift-drag of tree items.

**Comment by daspryJ** [comment]
Oh, thank you. I still don‘t understand the other ways the shift-drag option can be used, but if you made it work for my specific use case, I cannot complain!
~~I‘ll just wait until the next release maybe before marking it solved, because I don‘t really know how to update it directly from the github source code.~~
Nevermind, I updated the extention and it works! 
What would make it really nice is to create a hyperlink with the tab title as text. (extra would be to turn it into a list that follow the tree structure)
Right now from what I‘ve seen, it either pastes all the tree urls in plain text, or just the first tab in a hyperlink with the url as hypertext.

But it‘s still very nice, thank you again.

**Reply by piroor** [reply]
I recommend you to use [Copy Selected Tabs to Clipboard](https://addons.mozilla.org/firefox/addon/copy-selected-tabs-to-clipboar/). It provides ability to copy list of tabs in various formats including the tree structure.

**Reply by daspryJ** [reply]
Well, other than the fact that it requires another extension, it does somewhat copy the tree structure if I use the "Markdown(list)" option, but any option still copy the urls in plain text, not in a hyperlink form.
Moreover, even the "Markdown(list)" option only reproduce the tree structure up to two levels, then it just copy the 3rd, 4th etc levels as 2nd levels tabs.

To illustrate my main issue, right now it looks like this (each paragraph is a different option) :
![image](https://github.com/user-attachments/assets/f70d0bb3-0f01-4ee6-8374-e9eb4255b9fd)
And I‘m used to just have to drag and drop from the tree and have this : 
![image](https://github.com/user-attachments/assets/7f2eab75-3b13-4fea-82d6-24f31a3b66d4)

**Reply by piroor** [reply]
Hmm, currently the addon Copy Selected Tabs to Clipboard does not support generating nested HTML list. `%RT%%PREFIX(<ul>%EOL%)%%SUFFIX(</ul>%EOL%)%<li><a href="%URL_HTMLIFIED%">%TITLE_HTMLIFIED%</a></li>` will generate HTML list but it is flat.

**Reply by piroor** [reply]
I've filed an issue: https://github.com/piroor/copy-selected-tabs-to-clipboard/issues/56

**Reply by daspryJ** [reply]
Oh but your code does copy it in a hyperlink with the tab title as hypertext, that‘s already very good! It was my main concern, and I think that‘s the way most users would want it to be by default.
As for the nested HTML list, that would be the icing on the cake, but it‘s not deal breaking for me.
